dali/ft/filecopy.cpp

```diff
@@ -536,7 +546,8 @@ int FileSizeThread::run()
                 cur->size = thisSize;
                 break;
             }
-            if (copy==1) { // need to set primary
+            if (copy==1)
```

Comment:
The code below alters then potentially destroys cur, the other thread in wait() could be accessing it.
This needs protecting too.